client/src/featureform/client.py

```diff
@@ -188,6 +192,135 @@ def close(self):
         """
         self.impl.close()
 
+    def add_trigger(self, trigger, resource):
+        """
+        Add a trigger to a resource
+
+        **Example:**
+        ```py title="definitions.py"
+        client.add_trigger("trigger_name", ("resource_name", "resource_variant, resource_type"))
+        ```
+
+        Args:
+            trigger(Union[str, TriggerResource]): The name of the trigger
+            resource(Union[tuple, FeatureColumnResource, TrainingSetVariant]): The name, variant and type of the resource
+        """
+        req = metadata_pb2.TriggerRequest()
```

Comment:
nit: you could probably move all this logic to its own function since it's reused in both places

to_trigger_request(resource)